pkg/restapi/vc/operation/operations.go

```diff
@@ -178,53 +193,110 @@ func (c *Operation) verifyCredentialHandler(rw http.ResponseWriter, req *http.Re
 		Message:  message}
 
 	rw.WriteHeader(http.StatusOK)
-	c.writeResponse(rw, response)
+	o.writeResponse(rw, response)
 }
 
-func (c *Operation) createProfileHandler(rw http.ResponseWriter, req *http.Request) {
+func (o *Operation) createProfileHandler(rw http.ResponseWriter, req *http.Request) {
 	data := ProfileRequest{}
 	err := json.NewDecoder(req.Body).Decode(&data)
 
 	if err != nil {
-		c.writeErrorResponse(rw, http.StatusBadRequest, "Failed to write response for invalid request received")
+		o.writeErrorResponse(rw, http.StatusBadRequest, "Failed to write response for invalid request received")
 
 		return
 	}
 
-	profileResponse, err := c.createProfile(&data)
+	profileResponse, err := o.createProfile(&data)
 
 	if err != nil {
-		c.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
+		o.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
 
 		return
 	}
 
 	rw.WriteHeader(http.StatusCreated)
-	c.writeResponse(rw, profileResponse)
+	o.writeResponse(rw, profileResponse)
 }
 
-func (c *Operation) getProfileHandler(rw http.ResponseWriter, req *http.Request) {
-	vars := mux.Vars(req)
-	profileID := vars[profilePathVariable]
+func (o *Operation) getProfileHandler(rw http.ResponseWriter, req *http.Request) {
+	profileID := mux.Vars(req)["id"]
 
-	profileResponseJSON, err := c.profileStore.GetProfile(profileID)
+	profileResponseJSON, err := o.profileStore.GetProfile(profileID)
 	if err != nil {
 		if err == errProfileNotFound {
-			c.writeErrorResponse(rw, http.StatusNotFound, "Failed to find the profile")
+			o.writeErrorResponse(rw, http.StatusNotFound, "Failed to find the profile")
 
 			return
 		}
 
-		c.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
+		o.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
+
+		return
+	}
+
+	o.writeResponse(rw, profileResponseJSON)
+}
+
+func (o *Operation) storeVCHandler(rw http.ResponseWriter, req *http.Request) {
+	data := &CreateCrendentialResponse{}
+	err := json.NewDecoder(req.Body).Decode(&data)
+
+	if err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, "invalid request received")
+
+		return
+	}
+
+	if err = validateRequest(data.Profile, data.ID); err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
+
+		return
+	}
+
+	doc := operation.StructuredDocument{}
+	doc.Content = make(map[string]interface{})
+	doc.Content["message"] = data
+	doc.ID = data.ID[strings.LastIndex(data.ID, "/")+1:]
+
+	locationOfDocument, err := o.client.CreateDocument(data.Profile, &doc)
+	if err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
+
+		return
+	}
+
+	o.writeResponse(rw, locationOfDocument)
```

Comment:
I think this is writing the EDV location right? Does the caller of this API actually need that/should they have that info? Just double checking...I'm not actually sure myself

Reply:
As per the discussion with troy he metioned vc service is basically a wrapper over edv , it will have profile info crypto info extra rest api should behave as edv only. So for now i think its okay